### ISSUE TITLE
Kubernetes version and custom AMIs

### ISSUE THREAD

**Issue by atmosx** [awslabs/amazon-eks-ami#1480]
**Environment**:
- AWS Region: us-east-1
- Instance Type(s): c2a.2xlarge
- EKS Platform version: eks.10
- Kubernetes version: 1.24

tl;dr: 

Going through the [documentation](https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html#mng-ami-id-conditions) I can see that we don't have to specific `amiType`, `releaseVersion` and `version`.  When using custom AMIs with managed node groups, how can we control the kubernetes version running in the control plane and _other_ node groups?

In detail:

We have three node groups on each EKS cluster. The goal is to use a newly created AMI to an existing node group. The terraform configuration  we came up with:

```tf
resource "aws_eks_cluster" "cluster" {
  name     = local.settings.kubernetes.name
  ...
  version = local.settings.kubernetes.version #=> 1.24
  ...
}

resource "aws_launch_template" "abcd" {
  ...
  image_id = local.settings.kubernetes.pools.sample_app.ami_id #=> ami-abcd...
  ...
}

resource "aws_eks_node_group" "abcd" {
  ...
  ami_type = "CUSTOM"

  launch_template {
    id      = aws_launch_template.abcd.id
    version = aws_launch_template.abcd.latest_version
  }
  ...
}
```

Apparently to use a custom AMI we need to set `ami_type = "CUSTOM"` in terraform. However, once we've set that we got the net error message: 

```
 Message_: "You cannot specify a kubernetes version to use when specifying an image id to use within the launch template."
```

I'm trying to understand how/why the `version` option in the `aws_eks_cluster` resource is affected the state here. Looks like we can either have kubernetes clusters with custom and/or non AMIs but not both which doesn't make sense 🤔

**Comment by bryantbiggs** [comment]
setting the `release_version` and `version` on the managed nodegroup is not valid when using a custom AMI - these are details that the managed nodegroup uses in order to select the appropriate EKS AMI

You can see how we manage this here https://github.com/bryantbiggs/terraform-aws-eks/blob/51cc6bec880ac8dc361b60a4b05d5f2bcd98eb6a/modules/eks-managed-node-group/main.tf#L326-L329

**Comment by cartermckinnon** [comment]
Yes, this is a Terraform thing.
